cpp/src/Slice/SliceUtil.cpp

```diff
@@ -536,6 +536,17 @@ Slice::ciequals(const string& lhs, const string& rhs)
                                                                       });
 }
 
+TypePtr
+Slice::unwrapIfOptional(const TypePtr& type)
+{
+    auto result = type;
+    if (auto optional = OptionalPtr::dynamicCast(type))
+    {
+        result = optional->underlying();
+    }
+    return result;
+}
+
```

Comment:
Do we really need this intermediary variable `result` here?
Why not just
```
if (auto optional = OptionalPtr::dynamicCast(type))
{
    return optional->underlying();
}
return type;
```

Reply:
Fixed.